src/main/java/workflow/WorkflowManager.java

```diff
@@ -0,0 +1,20 @@
+package workflow;
+
+import workflow.context.AbstractCompositeContext;
+
+import java.util.HashMap;
+
+public enum WorkflowManager {
+    INSTANCE;
+
+    HashMap<String, AbstractCompositeContext> contextMap = new HashMap<String, AbstractCompositeContext>();
+
+    public AbstractCompositeContext getContextByKey(String key) {
```

Comment:
this is nothing more than a central repository to get at any context by just using a key.